doc/release/3.4.0.rst

```diff
@@ -0,0 +1,249 @@
+
+     .. code-block:: lua
+
+         box.cfg{memtx_use_mvcc_engine = true}
+
+   * Vinyl operations may yield implicitly, so a transaction may be aborted
+     with TRANSACTION_CONFLICT in case of concurrent transactions.
+
+.. _3-4-native:
+
+[EE] New boostrap strategy in failover: ``native``
+--------------------------------------------------
+
+Now supervised failover coordinator supports three bootstrap strategies:
+native, supervised, auto.
+
+The new ``native `` strategy acts more or less similar to the ``auto`` strategy,
+but relaxes its limitations. It is based on the supervised strategy and basically
+performs two things:
+* issues ``box.ctl.make_bootstrap_leader({graceful = true})`` to bootstrap
+  a replicaset;
+* issues ``box.ctl.make_bootstrap_leader()`` to keep the bootstrap leader
+  record pointing to the instance that is currently in the RW mode (to register
+  new replicas).
+
+To enable the ``native `` bootstrap strategy, set it in the ``replication`` section
+of the cluster's configuration, together with a proper failover strategy
+(for ``native``, you can choose any failover strategy you like, for example ``supervised``):
+
+.. code-block:: yaml
+
+    replication:
+      failover: supervised
+      bootstrap_strategy: native
+
+.. _3-4-runtime-priv:
+
+[CE] Runtime priviledges for ``lua_call`` granted before ``box.cfg()``
+----------------------------------------------------------------------
+
+It is now possible to grant execution privileges for Lua functions
+through the declarative configuration, even when the database is in
+read-only mode or has an outdated schema version. You might also
+permit ``guest`` to execute Lua functions before the initial bootstrap.
+
+You can specify function permissions using the ``lua_call`` option in
+the configuration, for example:
+
+.. code-block:: lua
+
+    credentials:
+      users:
+        alice:
+          privileges:
+            - permissions: [execute]
+              lua_call: [my_func]
+
+This grants the ``alice`` user permission to execute the ``my_func`` Lua
+function, regardless of the database's mode or status. The special option
+``lua_call: [all]`` is also supported, granting access to all global Lua
+functions except built-in ones, bypassing database restrictions.
+
+Privileges will still be written to the database when possible to
+maintain compatibility and consistency with other privilege types.
+
+[CE] New methods in the ``config`` module to check instance roles
+-----------------------------------------------------------------
+
+Three new methods are now available in the ``config`` module:
+
+* ``has_role(<role_name>, {instance = <instance_name})`` returns ``true`` if
+  the instance with the name ``<instance_name>`` has the role ``<role_name>``
+  enabled in the current configuration, or ``false`` if not.
+  The second argument is optional: if not provided, the check is performed
+  for the instance the method is called on.
```

Comment:
Nit: It feels a bit too formal for the announcement. Would look appropriate in an API documentation, but here I would make it simpler:

> `config:has_role('myrole')` tells whether the current instance has role `myrole`. `config:has_role('myrole', {instance = 'i-001'})` do the same for the given instance.

Same for `config:is_router()`, `config:is_storage()`.

Reply:
I'm OK.